services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -157,6 +156,7 @@ public void configure(Binder binder)
             binder.bind(DataSegmentArchiver.class).to(OmniDataSegmentArchiver.class).in(LazySingleton.class);
 
             binder.bind(ExecutorLifecycle.class).in(ManageLifecycle.class);
```

Comment:
This is needed, the LifecycleModule.register just ensures eager creation, but .in(ManageLifecycle) is needed to actually do the starting and stopping.